neo/io/nestio.py

```diff
@@ -431,11 +448,11 @@ def _get_selected_ids(self, gid, id_column, time_column, t_start, t_stop,
         Returns
         list of selected gids
         """
-        gid_ids = np.array([0, data.shape[0]])
+        gids = np.array([0, data.shape[0]])
         if id_column is not None:
-            gid_ids = np.array([np.searchsorted(data[:, 0], gid, side='left'),
-                                np.searchsorted(data[:, 0], gid, side='right')])
-        gid_data = data[gid_ids[0]:gid_ids[1], :]
+            gids = np.array([np.searchsorted(data[:, 0], gid, side='left'),
```

Comment:
Should not this be the searched in the ID column, i.e.,
```suggestion
            gids = np.array([np.searchsorted(data[:, id_column], gid, side='left'),
```

A similar change is probably needed below for `id_shifts`, i.e.,
```python
id_shifts[0] = np.searchsorted(gid_data[:, time_column],
```